tools/loader/ImageClassifier.cpp

```diff
@@ -413,22 +206,39 @@ static int processAndPrintResultsImpl(Tensor *SMT,
       applySoftmax(SH);
     }
 
-    auto topKPairs = getTopKPairs(SH);
-    printTopKPairs(topKPairs);
-    if (!expectedMatchingLabels.empty()) {
-      retVal +=
-          checkExpectedLabel(topKPairs, fileName, expectedMatchingLabels[i]);
+    if (topKCount) {
+      auto topKPairs = getTopKPairs(SH);
+      printTopKPairs(topKPairs);
+      if (!expectedMatchingLabels.empty()) {
+        retVal +=
+            checkExpectedLabel(topKPairs, fileName, expectedMatchingLabels[i]);
+      }
     }
   }
 
   return retVal;
 }
 
+class ImageClassifierProcessResult : public PostProcessOutputDataExtension {
+public:
+  int processOutputs(
+      const llvm::StringMap<Placeholder *> &PHM, PlaceholderBindings &bindings,
+      llvm::ArrayRef<std::string> inputImageBatchFilenames) override;
+};
+
 /// Given the output Softmax Tensor \p SMT and \p functionName, switch between
 /// the correct element type to print the results of inference as contained in
 /// \p SMT, \returns the number of found mismatches.
-static int processAndPrintResults(Tensor *SMT,
-                                  llvm::ArrayRef<std::string> imageList) {
+int ImageClassifierProcessResult::processOutputs(
+    const llvm::StringMap<Placeholder *> &PHM, PlaceholderBindings &bindings,
+    llvm::ArrayRef<std::string> imageList) {
+  if (PHM.size() > 1) {
+    LOG(INFO) << "Network has more then one output. Process results not "
```

Comment:
Right now in image classifier there is this:

```
if (!profilingGraph()) {
        std::lock_guard<std::mutex> lock(ioMu);
        numErrors += processAndPrintResults(bindings.get(outputPH),
                                            inputImageBatchFilenames);
}
```

I believe this was added there because of object detection profiling:
https://github.com/pytorch/glow/pull/3766

Now in refactoring I didn't want to have that check there because it needs to be generic, and down the road a real object-detection app can be build on top if it, but I didn't want to break current functionality. Until app is up-streamed people will use image-classifier for object detection. So I made it as a LOG info and return 0. That was kind of my thinking. Not 100% if it's best way to address this.

Reply:
Got it. So if we're trying to maintain current functionality then would it make more sense to check `profilingGraph()` here to skip `processAndPrintResultsImpl()` and return 0? And then we could also check the size after and `LOG(FATAL)` if so? WDYT?